### PR TITLE
[FEATURE] Add invoice api call

### PR DESCRIPTION
**Pull request by tradeweeyoung**

### CODE REVIEW

**Comment by levady** [comment]
LGTM

**Comment by tradeweeyoung** [comment]
@levady , need to change version #?

File: lib/zuora/invoice.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Zuora
+  class Transaction < Resource; end
```
Comment (gherry) [inline lib/zuora/invoice.rb]: Is there a reason for this `Transaction` class?

File: lib/zuora/invoice.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  class Invoice < Transaction
+    class << self
+      def find_by_account_id(account_id, options={})
+        Zuora.request(:get, find_by_account_id_base_resource_url(account_id), options)
+      end
+
+      def find_by_account_id_base_resource_url(account_id)
+        [superclass.base_resource_url, "invoices/accounts", account_id].join("/")
```
Comment (gherry) [inline lib/zuora/invoice.rb]: ah okay, I see, you added `Transaction` just to construct the url. I would rather either hardcoding the path for now or having the following structure

```ruby
module Zuora
  module Transaction
    class Invoice
    end
  end
end
```

**Comment by levady** [comment]
@tradeweeyoung Yes, we need to bump the version to 0.0.3 so the main app will pick it automatically when we do bundle install.

**Pull request merged**